src/app/unlocked-stage/unlocked-stage.component.ts

```diff
@@ -1,156 +1,331 @@
+    const that = this;
+    this.$net.mouseenter(function () {
+      that._renderer.listen(that._canvas, 'mousemove', (event) => that.moveOverCanvasNet(event))
+    });
+  }
+
+  measureSpray() {
+    const that = this;
+    setTimeout(function () {
+      if (that._mousedown) {
+        that.stopAttack();
+        that.sprayValue -= 4;
+        that.measureSpray();
+      }
+    }, 500)
+  }
+
+  attack() {
+    this.$spray.attr('src', '../assets/img/unlocked-stage/spray.gif');
+    this._mousedown = true;
+    this.measureSpray();
+  }
+
+  stopAttack() {
+    this.$spray.attr('src', '../assets/img/unlocked-stage/spray-cursor.gif');
+    this._mousedown = false;
+  }
+
+  moveOverCanvasSpray(e) {
+    this._canvas.style.cursor = 'none';
+    this._spray.css({display: 'block'});
+    this._spray.css({left: e.pageX - 510 + 'px'});
+    this._spray.css({top: e.pageY - 150 + 'px'});
+  }
+
+  moveOverCanvasNet(e) {
+    if (!(e.pageX > 1270 || e.pageX < 1200 || e.pageY < 517)) {
+      this.$net.css({top: e.pageY - 150 + 'px'});
     }
+  }
 
-    leaveCanvas() {
-        this._spray.style.display = 'none';
+  leaveCanvas() {
+    if (this._spray) {
+      this._spray.css({display: 'none'})
     }
+  }
 
-    animateDiv() {
-        const newMosquitoPos = this.makeNewPosition();
-        const oldMosquitoTop = this.mosquito.offsetTop;
-        const oldMosquitoLeft = this.mosquito.offsetLeft;
-        const currObj = this;
-        const speed = this.calcSpeed([oldMosquitoTop, oldMosquitoLeft], newMosquitoPos);
-        $('img[name=animate]').animate({ top: newMosquitoPos[1], left: newMosquitoPos[0] }, speed, function() {
-            if (currObj._mosquitoBarWidth <= 0) {
+  animateDiv(mosquito, newMosquitoPosition, id) {
+    let collisionWithSpray: boolean;
+    const newMosquitoPos = newMosquitoPosition;
+    const oldMosquitoTop = mosquito.offset().top;
+    const oldMosquitoLeft = mosquito.offset().left;
+    const currObj = this;
+    const speed = this.calcSpeed([oldMosquitoTop, oldMosquitoLeft], newMosquitoPos);
+    mosquito.animate({top: newMosquitoPos[1], left: newMosquitoPos[0]}, {
+      duration: speed,
+      start: function () {
+        collisionWithSpray = false;
+      },
+      step: function () {
+        if (currObj.$net && currObj.$net.css('display') === 'block' && currObj.checkCollison(mosquito, currObj.$net)) {
+          if (!currObj._collisionWithNetStatus) {
+            const x = mosquito.offset().left;
+            const y = mosquito.offset().top;
+            mosquito.stop();
+            setTimeout(function () {
+              const newNextMosquitoPosition = [x - 1000, y];
+              currObj.animateDiv(mosquito, newNextMosquitoPosition, id);
+            }, 1000);
+            currObj._collisionWithNetStatus = true;
+          }
+        }
+        if ((currObj.checkCollison(mosquito, currObj.$human))) {
+          if (!currObj._collisionWithManStatus) {
+            currObj._humanBarWidth -= 10;
```

Comment:
I think the human health reduction can be made 3 or 5, this way the human won't lose easily. Also make this a constant

Reply:
Done.